dagshub/data_engine/model/query_result.py

```diff
@@ -401,6 +404,74 @@ def download_binary_columns(
+
+        Args:
+            open_project: Automatically open the Label Studio project in the browser
+            ignore_warning: Suppress the prompt-warning if you try to annotate too many datapoints at once.
+            fields_to_embed: list of meta-data columns that will show up in Label Studio UI.
+             if not specified all will be displayed.
+            fields_to_exclude: list of meta-data columns that will not show up in Label Studio UI
+            repo: repository to extract the model from
+            name: name of the model in the mlflow registry
+            version: (optional, default: 'latest') version of the model in the mlflow registry
+            pre_hook: (optional, default: identity function) function that runs before datapoint is sent to the model
+            post_hook: (optional, default: identity function) function that converts mlflow model output
+            to the desired format
+            batch_size: (optional, default: 1) function that sets batch_size
+            metadata_column: (optional, default: 'prediction') write prediction results to metadatalogged in data engine.
+            If None, returns predictions.
+            return_predictions: (optional, default: True) returns predictions logged
+        Returns:
+            The URL of the created Label Studio workspace
+        """
+        if not metadata_column and not return_predictions:
+            raise ValueError("There is nothing to do. Either `metadata_column` or return_predictions` must be set!")
+
+        class TinyDL:
+            def __init__(self, dset, batch_size):
+                self.dset = dset
+                self.batch_size = batch_size
+
+            def __iter__(self):
+                self.curr_idx = 0
+                return self
+
+            def __next__(self):
+                self.curr_idx += self.batch_size
+                return [self.dset[idx] for idx in range(self.curr_idx - self.batch_size, self.curr_idx)]
+
+        init(*repo.split("/")[::-1])
+        model = mlflow.pyfunc.load_model(f"models:/{name}/{version}")
+        dset = DagsHubDataset(self, tensorizers=[lambda x: x])
+
+        predictions = []
+        for idx, local_paths in enumerate(TinyDL(dset, batch_size) if batch_size != 1 else dset):
+            for prediction, remote_path in zip(
+                post_hook(model.predict(pre_hook(local_paths))),
+                [result.path for result in self[idx * batch_size: (idx + 1) * batch_size]],
+            ):
+                predictions.append({remote_path: prediction})
+
+        if not metadata_column:
+            return predictions
+        with self.datasource.metadata_context() as ctx:
+            for remote_path in predictions:
+                ctx.update_metadata(remote_path, {metadata_column: predictions[remote_path]})
+        return predictions if return_predictions else True
```

Comment:
Throw in a couple `log_message()`s here, since this is a very interactive function that also might have to wait a long time to wait for the predicting to finish.

If you want to go all the way, you can make it a progress bar with rich, metadata uploading has a working progress you can look at